### PR TITLE
chore: add relevant workflows

### PR DESCRIPTION
**Pull request by Vvaradinov**
This PR adds the relevant CI workflows from Evmos/Ethermint. Skipping Android tests for the time being until the proper setup is made.

### CODE REVIEW

**Comment by Vvaradinov** [comment]
@fedekunze can you check the settings of this repo and enable the dependency graph? The CI run is giving me this error `Please ensure that Dependency graph is enabled along with GitHub Advanced Security on private repositories`

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
+name: Build
+on:
+  pull_request:
+    branches:
+      - main
```
Comment (fedekunze) [inline .github/workflows/build.yml:5]: @Vvaradinov needs to be changed to `master`
```suggestion
      - master
```

**Review by fedekunze — COMMENTED** [review]
Needs changes as the default branch is `master`. We also need to set up the `codeql.yaml` since it's not enabled by default

File: .github/workflows/lint.yml [hunk 2]
```diff
@@ -0,0 +1,72 @@
+name: Lint
+# Lint runs golangci-lint over the entire ethermint repository This workflow is
+# run on every pull request and push to main The `golangci` will pass without
+# running if no *.{go, mod, sum} files have been changed.
+on:
+  pull_request:
+  push:
+    branches:
+      - main
```
Comment (fedekunze) [inline .github/workflows/lint.yml:9]:
```suggestion
      - master
```

File: .github/workflows/markdown-links.yml [hunk 3]
```diff
@@ -0,0 +1,29 @@
+name: Check Markdown links
+on: 
+  pull_request:
+    paths:
+      - '**.md'
+  push:
+    branches:
+      - main
```
Comment (fedekunze) [inline .github/workflows/markdown-links.yml:8]:
```suggestion
      - master
```

File: .github/workflows/security.yml [hunk 4]
```diff
@@ -0,0 +1,37 @@
+name: Run Gosec
+on:
+  pull_request:
+  push:
+    branches:
+      - main
```
Comment (fedekunze) [inline .github/workflows/security.yml:6]:
```suggestion
      - master
```

File: .github/workflows/super-linter.yml [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+# For more information, see:
+# https://github.com/github/super-linter
+---
+name: Lint Code Base
+
+on:
+  push:
+    branches: ["main"]
+  pull_request:
+    branches: ["main"]
```
Comment (fedekunze) [inline .github/workflows/super-linter.yml:14]:
```suggestion
    branches: ["master"]
  pull_request:
    branches: ["master"]
```

File: .github/workflows/super-linter.yml [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+          LINTER_RULES_PATH: /
+          YAML_CONFIG_FILE: .yamllint
+          VALIDATE_ALL_CODEBASE: false
+          MARKDOWN_CONFIG_FILE: .markdownlint.yml
+          PROTOBUF_CONFIG_FILE: .protolint.yml
+          VALIDATE_NATURAL_LANGUAGE: false
+          VALIDATE_OPENAPI: false
+          VALIDATE_JSCPD: false
+          VALIDATE_GO: false
+          DEFAULT_BRANCH: "main"
```
Comment (fedekunze) [inline .github/workflows/super-linter.yml:37]:
```suggestion
          DEFAULT_BRANCH: "master"
```

File: .github/workflows/test.yml [hunk 7]
```diff
@@ -0,0 +1,35 @@
+name: Tests
+on:
+  pull_request:
+  push:
+    branches:
+      - main
```
Comment (fedekunze) [inline .github/workflows/test.yml:6]:
```suggestion
      - master
```

File: .github/workflows/test.yml [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+      - release/**
+
+jobs:
+  cleanup-runs:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: rokroskar/workflow-run-cleanup-action@master
+        env:
+          GITHUB_TOKEN: "${{ secrets.GITHUB_TOKEN }}"
+    if: "!startsWith(github.ref, 'refs/tags/') && github.ref != 'refs/heads/main'"
```
Comment (fedekunze) [inline .github/workflows/test.yml:16]:
```suggestion
    if: "!startsWith(github.ref, 'refs/tags/') && github.ref != 'refs/heads/master'"
```

**Review by fedekunze — CHANGES_REQUESTED** [review]

**Review by fedekunze — APPROVED** [review]

**Comment by fedekunze** [comment]
@mergifyio backport release/1.10

**Pull request merged**